### DISCUSSION TITLE
Blocking Traffic to the Cloud after setup completed

### DISCUSSION THREAD

**Discussion by crazycoder1999** [jasonacox/tinytuya#79] [General]
Hi,
 I would like to try tinytuya! great project!
I read some of the instruction and I was thinking to use a raspberry pi and some script with tinytuya to control my devices.
Q: Once the tinytuya works can I block all the traffic to the tuya cloud and continue to control my devices locally or is there some kind of keepalive mechanism with the cloud?
Anyone tried a configuration like that? which address were blocked? 
Thanks you

**Comment by jasonacox** [comment]
Yes, once you have the Tuya device "Local Keys" you can disconnect them from the internet and continue to read and control them via `TinyTuya` and your local network.  I have not tried this but some have reported blocking outbound access based on MAC address or moving their Tuya devices to an IoT VLAN on their network that does not have internet access.  Based on my understanding, there are a few things with this setup you should be aware of after disabling internet:
* Integration with Alexa and Google will no longer work in this setup as these use the Tuya Cloud for communication
* Firmware updates may not happen but the _SmartLife_ app should still be able to talk to these devices on the local network.
* You will not be able to pair and get the "Local Key" unless the devices can talk to the Tuya Cloud
* Some of the Tuya devices only operate with the Tuya Cloud, primarily those that are Sensors or battery operated.

Others may have more details, advice or suggestions.  If you have success, please post what you did!

**Reply by crazycoder1999** [reply]
Thanks you for the quick response @jasonacox !
I will try as soon as I have time to try this setup.
For Alexa integration I think I can create a fake device using a raspberry pi as a bridge.
I keep you informed, thx
